src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/AsylumCasesRetriever.java

```diff
@@ -0,0 +1,150 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.joining;
+import static org.springframework.http.HttpHeaders.*;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.State.APPEAL_STARTED;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.State.UNKNOWN;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.State;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.security.AccessTokenDecoder;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.security.IdamAuthorizor;
+
+@Service
+class AsylumCasesRetriever {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AsylumCasesRetriever.class);
+
+    private static final String PATH_TEMPLATE = "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases";
+    private static final String PAGINATION_PATH_TEMPLATE = "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/pagination_metadata";
+    private static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    private final String ccdBaseUrl;
+    private final String testSystemUser;
```

Comment:
My bad for not thinking it through, but can we remove `test` from these variables -- this is not a test :)

Reply:
done